src/Interaction/CommandInteraction.php

```diff
@@ -61,14 +61,35 @@ public function deleteInteractionResponse(): ExtendedPromiseInterface
         );
     }
 
-    public function getOption($option): ?OptionStructure
+    public function getOption(string $path): ?OptionStructure
     {
-        return $this->options[$option] ?? null;
+        $segments = explode('.', $path);
+        return $this->findOption($this->options, $segments);
     }
 
-    public function hasOption($option): bool
+    private function findOption(array $options, array $segments): ?OptionStructure
     {
-        return isset($this->options[$option]);
+        $currentSegment = array_shift($segments);
+
+        foreach ($options as $opt) {
+            if ($opt->name === $currentSegment) {
+                if (empty($segments)) {
+                    return $opt;
+                }
+
+                if (!empty($opt->options)) {
+                    return $this->findOption($opt->options, $segments);
+                }
+            }
+        }
```

Comment:
I think it'd be more readable to use [array_find](https://packagist.org/packages/freezemage0/array_find) here, the package is already installed in the repo, would just need to be imported in the file (php 8.4 will have this function natively, so I'll worry about phasing out the package eventually)

Then it'd be something like this instead:
```php
$option = array_find($options, fn (OptionStructure $option) => $option->name === $currentSegment);

if (empty($segments)) {
    return $option;
}

return empty($option->options) ? null : $this->findOption($option->options, $segments);
```

Less nesting generally has my preference :smile: